### PR TITLE
[NLU-3965] - Catalan N of N dates 

### PR DESCRIPTION
**Pull request by Conor-Keaney**
Added Catalan written dates and dates as numbers for the use cases:

1. l'u de gener de 2000
2. el dia 1 de gener de l'any 2000
3. l'u de gener de l'any 2000
4. u de l'u del 2001

https://inindca.atlassian.net/browse/NLU-3965

### CODE REVIEW

**Review by BobbyMartins5 — APPROVED** [review]

File: Python/libraries/recognizers-date-time/recognizers_date_time/resources/catalan_date_time.py [hunk 1]
```diff
@@ -20,6 +22,7 @@ class CatalanDateTime:
     OfPrepositionRegex = f'(\\bde\\b)'
     TwoDigitYearRegex = f'\\b(?<![$])(?<year>([0-9]\\d))(?!(\\s*((\\:\\d)|{AmDescRegex}|{PmDescRegex}|\\.\\d))|\\.?[º°ª])\\b'
     WrittenOneToNineRegex = f'(?:u(n)?|dos|tres|quatre|cinc|sis|set|vuit|nou)'
+    WrittenOneToTwelveRegex = f'(?<month>u(n)?|dos|tres|quatre|cinc|sis|set|vuit|nou|onze|dotze)'
```
Comment (srandal) [inline Python/libraries/recognizers-date-time/recognizers_date_time/resources/catalan_date_time.py]: Change `u(n)?` to `un?` to be consistent with WrittenDayRegex, and maybe make the same change to WrittenOneToNineRegex and OclockRegex.

File: Python/libraries/recognizers-date-time/recognizers_date_time/resources/catalan_date_time.py [hunk 2]
```diff
@@ -20,6 +22,7 @@ class CatalanDateTime:
     OfPrepositionRegex = f'(\\bde\\b)'
     TwoDigitYearRegex = f'\\b(?<![$])(?<year>([0-9]\\d))(?!(\\s*((\\:\\d)|{AmDescRegex}|{PmDescRegex}|\\.\\d))|\\.?[º°ª])\\b'
     WrittenOneToNineRegex = f'(?:u(n)?|dos|tres|quatre|cinc|sis|set|vuit|nou)'
+    WrittenOneToTwelveRegex = f'(?<month>un?|dos|tres|quatre|cinc|sis|set|vuit|nou|onze|dotze)'
```
Comment (andrew-gradinari) [inline Python/libraries/recognizers-date-time/recognizers_date_time/resources/catalan_date_time.py]: Is "deu" (ten) missing from this regex?

File: Python/libraries/recognizers-date-time/recognizers_date_time/resources/catalan_date_time.py [hunk 3]
```diff
@@ -11,7 +13,7 @@ class CatalanDateTime:
     RangePrefixRegex = f'(des de|entre)'
     DayRegex = f'\\b(?<day>01|02|03|04|05|06|07|08|09|10|11|12|13|14|15|16|17|18|19|1|20|21|22|23|24|25|26|27|28|29|2|30|31|3|4|5|6|7|8|9)(?:\\.[º°])?(?=\\b|t)'
     MonthNumRegex = f'(?<month>1[0-2]|(0)?[1-9])\\b'
-    WrittenDayRegex = f'(?<day>un|dos|tres|quatre|cinc|cinc|sis|set|vuit|nou|nou|deu|onze|dotze|dotze|tretze|catorze|quinze|setze|mor disset|mor divuit|dinnou|vint|vint-i-un|vint-i-dos|vint-i-tres|vint-i-quatre|vint-i-cinc|vint-i-sis|vint-i-set|vint-i-vuit|vint-i-nou|trenta|trenta-un)'
+    WrittenDayRegex = f'(?<day>un?|dos|tres|quatre|cinc|cinc|sis|set|vuit|nou|nou|deu|onze|dotze|dotze|tretze|catorze|quinze|setze|mor disset|mor divuit|dinnou|vint|vint-i-un|vint-i-dos|vint-i-tres|vint-i-quatre|vint-i-cinc|vint-i-sis|vint-i-set|vint-i-vuit|vint-i-nou|trenta|trenta-un)'
```
Comment (andrew-gradinari) [inline Python/libraries/recognizers-date-time/recognizers_date_time/resources/catalan_date_time.py]: Looks like there are duplicates for "cinc", "nou", and "dotze."
Google Translate also gives me some other translations:
- "disset" for seventeen
- "divuit" for eighteen
- "dinou" for nineteen
- "vint-i-u" / "vint-i-un" for twenty-one
- "trenta-u" / "trenta-un" for thirty-one

Could you confirm these are correct?

**Review by srandal — APPROVED** [review]

**Review by andrew-gradinari — APPROVED** [review]

**Review by kevinwalshgen — APPROVED** [review]
Just ensure that there weren't any existing tests that could have been un-skipped

**Review by colm-dillon — APPROVED** [review]

**Review by KanchanKumar12 — APPROVED** [review]

**Pull request merged**